### PR TITLE
[TF-196] autocomplete crasher when satisfying diff protocol req

### PR DESCRIPTION
**Pull request by marcrasi**
The autocompleter was running TypeCheckProtocol with NULL ParameterIndices.

Resolves https://bugs.swift.org/browse/TF-196.

### CODE REVIEW

File: include/swift/AST/Attr.h [hunk 1]
```diff
@@ -1503,6 +1503,8 @@ class DifferentiableAttr final
   void setVJPFunction(FuncDecl *decl) { VJPFunction = decl; }
 
   bool parametersMatch(const DifferentiableAttr &other) const {
+    assert(ParameterIndices);
+    assert(other.ParameterIndices);
```
Comment (dan-zheng) [inline include/swift/AST/Attr.h]: Merge into one line?
```suggestion
    assert(ParameterIndices && other.ParameterIndices);
```

**Review by dan-zheng — APPROVED** [review]
Nice IDE test!

**Review by rxwei — APPROVED** [review]

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

File: include/swift/AST/Attr.h [hunk 2]
```diff
@@ -1503,6 +1503,8 @@ class DifferentiableAttr final
   void setVJPFunction(FuncDecl *decl) { VJPFunction = decl; }
 
   bool parametersMatch(const DifferentiableAttr &other) const {
+    assert(ParameterIndices);
```
Comment (dan-zheng) [inline include/swift/AST/Attr.h]: Oops - it seems this condition is not needed anymore?

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow macos

**Pull request merged**